From 8da245bac17a4f7e444446619ac9a74ca61282d7 Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Mon, 23 Jun 2025 17:38:00 +0800
Subject: [PATCH 104/123] mtk: mt76: mt7996: support informing the fw that peer
 sta is mtk sta

Informing the underlying fw that the peer station is mtk sta by setting
the BIT(3) of extra_info of STA REC BASIC tlv.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>
---
 mt76_connac_mcu.c | 3 +++
 mt76_connac_mcu.h | 1 +
 2 files changed, 4 insertions(+)

diff --git a/mt76_connac_mcu.c b/mt76_connac_mcu.c
index edeca712..2006b2b9 100644
--- a/mt76_connac_mcu.c
+++ b/mt76_connac_mcu.c
@@ -386,6 +386,9 @@ void mt76_connac_mcu_sta_basic_tlv(struct mt76_dev *dev, struct sk_buff *skb,
 		basic->extra_info |= cpu_to_le16(EXTRA_INFO_NEW);
 	basic->conn_state = conn_state;
 
+	if (link_sta && link_sta->sta->vendor_mtk)
+		basic->extra_info |= cpu_to_le16(EXTRA_INFO_MTK_STA);
+
 	if (!link_sta) {
 		basic->conn_type = cpu_to_le32(CONNECTION_INFRA_BC);
 
diff --git a/mt76_connac_mcu.h b/mt76_connac_mcu.h
index 38b483ba..17df0f90 100644
--- a/mt76_connac_mcu.h
+++ b/mt76_connac_mcu.h
@@ -296,6 +296,7 @@ struct sta_rec_basic {
 	u8 peer_addr[ETH_ALEN];
 #define EXTRA_INFO_VER	BIT(0)
 #define EXTRA_INFO_NEW	BIT(1)
+#define EXTRA_INFO_MTK_STA BIT(3)
 	__le16 extra_info;
 } __packed;
 
-- 
2.45.2

